apps/src/pythonlab/pyodideRunner.ts

```diff
@@ -144,3 +151,17 @@ function isNeighborhoodLevel() {
       ?.name === MiniApps.Neighborhood
   );
 }
+
+function handleRunEndedUnexpectedly(
+  consoleManager: ConsoleManager | null,
+  message: string
+) {
+  consoleManager?.writeConsoleMessage(getSystemMessage(message, appName));
+  if (isNeighborhoodLevel()) {
+    CodebridgeRegistry.getInstance().getNeighborhood()?.reset();
+    CodebridgeRegistry.getInstance().getNeighborhood()?.onRun();
```

Comment:
Could you explain why we run neighborhood in `handleRunEndedUnexpectedly`. I see this function is called when there's no source code or no `main.py`.

Reply:
It's because the neighborhood handles flipping the stop button back to run when it's done, so we need to start and stop it in order to get the neighborhood to process the signal to flip the run button back. I can add a comment.

Reply:
Oh gotcha! That makes sense. Thanks for the explanation and adding a comment!